gradle-palantir-java-format/src/main/groovy/com/palantir/javaformat/gradle/PalantirJavaFormatPlugin.java

```diff
@@ -16,42 +16,77 @@
 
 package com.palantir.javaformat.gradle;
 
+import com.palantir.javaformat.bootstrap.NativeImageFormatterService;
 import com.palantir.javaformat.java.FormatterService;
 import java.io.IOException;
+import java.nio.file.Path;
 import org.gradle.api.DefaultTask;
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.tasks.Input;
 import org.gradle.api.tasks.TaskAction;
 
 public final class PalantirJavaFormatPlugin implements Plugin<Project> {
 
     @Override
     public void apply(Project project) {
         project.getRootProject().getPlugins().apply(PalantirJavaFormatProviderPlugin.class);
+        project.getRootProject().getPlugins().apply(NativeImageFormatProviderPlugin.class);
         project.getRootProject().getPlugins().apply(PalantirJavaFormatIdeaPlugin.class);
 
         project.getPlugins().apply(PalantirJavaFormatSpotlessPlugin.class);
 
         project.getPlugins().withId("java", p -> {
-            project.getTasks().register("formatDiff", FormatDiffTask.class);
 
             // TODO(dfox): in the future we may want to offer a simple 'format' task so people don't need to use
             // spotless to try out our formatter
+            project.getTasks().register("formatDiff", FormatDiffTask.class, task -> {
+                if (NativeImageFormatProviderPlugin.isNativeImageSupported()
+                        && NativeImageFormatProviderPlugin.shouldUseNativeImage(project)) {
+                    task.getNativeImage().set(getNativeImplConfiguration(project));
+                }
+            });
         });
     }
 
-    public static class FormatDiffTask extends DefaultTask {
+    private static Provider<Path> getNativeImplConfiguration(Project project) {
+        return (project.getRootProject()
+                .getConfigurations()
+                .named(NativeImageFormatProviderPlugin.NATIVE_CONFIGURATION_NAME)
+                .map(conf -> conf.getSingleFile().toPath()));
+    }
+
+    public abstract static class FormatDiffTask extends DefaultTask {
+
+        private static Logger log = Logging.getLogger(FormatDiffTask.class);
+
+        @org.gradle.api.tasks.Optional
+        @Input
+        abstract Property<Path> getNativeImage();
```

Comment:
This should really be an `@InputFile` - can it be a `@Optional @InputFile RegularFileProperty`? If there's problems with the optional, you can do the trick of having a `ConfigurableFileCollection` that has one or more files.